components/TextFields/src/MDCInputTextField.h

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+/**
+ A set of Contained Input View states outlined in the Material guidelines. These states overlap with
+ and extend UIControlState.
+ */
+typedef NS_OPTIONS(NSInteger, MDCContainedInputViewState) {
```

Comment:
Given the significant overlap with UIControlState - are we providing additional APIs for things like `mdc_setEnabled:` and `mdc_setActivated:` and such?

Reply:
The way I originally envisioned it this notion of state was computed and readonly. It's value is discerned from the values of UIControl's `isEnabled` and `isSelected`, UITextField's `isEditing`, and the MDCInputTextField's `isSelected` and `isActivated`. So I do not think there should be a setter for enabled. And based off our revelation from earlier, I am beginning to think activated just shouldn't be a thing.